webapp/src/main/java/com/box/l10n/mojito/monitoring/PrefixHierarchicalNameMapper.java

```diff
@@ -1,45 +1,74 @@
 package com.box.l10n.mojito.monitoring;
 
-import com.google.common.base.Preconditions;
+
 import com.google.common.base.Strings;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
 import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Tag;
 import io.micrometer.core.instrument.config.NamingConvention;
 import io.micrometer.core.instrument.util.HierarchicalNameMapper;
 
-import java.util.stream.Collectors;
+import static com.google.common.base.Preconditions.checkNotNull;
 
 public class PrefixHierarchicalNameMapper implements HierarchicalNameMapper {
 
     String namePrefix;
     String tagKeyPrefix;
+    String stripCharacters;
     NamingConvention nameConventionOverride;
 
-    public PrefixHierarchicalNameMapper(String namePrefix, String tagKeyPrefix, NamingConvention nameConventionOverride) {
-        Preconditions.checkNotNull(namePrefix, tagKeyPrefix, nameConventionOverride);
-        this.namePrefix = namePrefix;
-        this.tagKeyPrefix = tagKeyPrefix;
+    public PrefixHierarchicalNameMapper(String namePrefix,
+                                        String tagKeyPrefix,
+                                        String stripCharacters,
+                                        @Nullable NamingConvention nameConventionOverride) {
+        this.namePrefix = checkNotNull(namePrefix);
+        this.tagKeyPrefix = checkNotNull(tagKeyPrefix);
+        this.stripCharacters = checkNotNull(stripCharacters);
         this.nameConventionOverride = nameConventionOverride;
     }
 
+    public PrefixHierarchicalNameMapper(String namePrefix,
+                                        String tagKeyPrefix,
+                                        NamingConvention nameConventionOverride) {
+        this(namePrefix, tagKeyPrefix, "", nameConventionOverride);
+    }
+
+    public PrefixHierarchicalNameMapper(String namePrefix,
+                                        String tagKeyPrefix) {
+        this(namePrefix, tagKeyPrefix, "", null);
+    }
+
     /**
      * Pretty much a copy of {@link HierarchicalNameMapper.DEFAULT} with prefixes added.
      */
     @Override
     public String toHierarchicalName(Meter.Id id, NamingConvention namingConvention) {
 
-        if (nameConventionOverride != null) {
-            namingConvention = nameConventionOverride;
-        }
+        namingConvention = Optional.ofNullable(nameConventionOverride).orElse(namingConvention);
```

Comment:
is there a real benefit of this notation in this case?

Reply:
IMHO, it improves readability quite a bit

Reply:
yeah, readability is a matter of taste and I have the opposite opinion here for example. So it seems to be an unnecessary change for this PR.